src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,212 @@
+
+                // Add comma if not last column. example: {"id":"1234","name":"Big Company"}
+                // the below ensures there is a comman after id but not after name.
+                if (i != structure.Columns.Count)
+                {
+                    stringAgg += ",";
+                }
+            }
+
+            if (structure.IsListQuery)
+            {
+                // Array wrappers if we are trying to get a list of objects.
+                columns = $"COALESCE(\'[\'+STRING_AGG(\'{{{stringAgg}}}\',', ')+\']\',\'[]\')";
+            }
+            else
+            {
+                columns = $"STRING_AGG(\'{{{stringAgg}}}\',', ')";
+            }
+
+            // the fromSql generates the query that will return the actual results that will be jsonified
+            // in format of earlier code.
+            string fromSql = $"{BuildSqlQuery(structure)}";
+            string query = $"SELECT {columns}"
+                + $" FROM ({fromSql}) AS {QuoteIdentifier(structure.SourceAlias)}";
+            return query;
+
+        }
+
+        /// <summary>
+        /// Build internal query for DW.
+        /// This will generate the query that will return results in sql.
+        /// Results of this query will be jsonified and returned to the user.
+        /// Sample: Get title, publishers and authors of a book.
+        /// SELECT TOP 100 [table0].[title] AS[title],
+        /// JSON_QUERY([table1_subq].[data]) AS[publishers],
+        /// JSON_QUERY(COALESCE('['+[table7_subq].[data]+']', '[]')) AS[authors]
+        /// FROM dbo_books AS[table0]
+        /// OUTER APPLY(SubQuery generated by recursive call to build function, will create the _subq tables)
+        /// </summary>
+        private string BuildSqlQuery(SqlQueryStructure structure)
+        {
+            string dataIdent = QuoteIdentifier(SqlQueryStructure.DATA_IDENT);
+            string fromSql = $"{QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier(structure.DatabaseObject.Name)} " +
+                             $"AS {QuoteIdentifier($"{structure.SourceAlias}")}{Build(structure.Joins)}";
+
+            fromSql += string.Join(
+                    "",
+                    structure.JoinQueries.Select(
+                        x => $" OUTER APPLY ({Build(x.Value)}) AS {QuoteIdentifier(x.Key)}({dataIdent})"));
+            string predicates = JoinPredicateStrings(
+                                    structure.GetDbPolicyForOperation(EntityActionOperation.Read),
+                                    structure.FilterPredicates,
+                                    Build(structure.Predicates),
+                                    Build(structure.PaginationMetadata.PaginationPredicate));
+            string columns = WrappedColumns(structure);
+            string orderBy = $" ORDER BY {Build(structure.OrderByColumns)}";
+
+            string query = $"SELECT TOP {structure.Limit()} {columns}"
+                + $" FROM {fromSql}"
```

Comment:
Is TOP the most performant for DW ?

Reply:
Based on review with UC, this seemed to be ok. We are waiting on contact from DW team. any perf related improvements will likely come in following pr's as we keep enhancing the query.